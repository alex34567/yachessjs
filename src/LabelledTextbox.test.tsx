// Setup code adapted from react docs
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from '@testing-library/react'
import LabelledTextBox from './LabelledTextbox'
import React from 'react'

let container: HTMLDivElement | null = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container!)
  container!.remove()
  container = null
})

test('Textbox without class', () => {
  act(() => {
    render(<LabelledTextBox label='Hi' type='text'/>, container)
  })
  expect(container).toMatchSnapshot()
})

test('Textbox with class', () => {
  act(() => {
    render(<LabelledTextBox className='Test' label='Hi' type='text'/>, container)
  })
  expect(container).toMatchSnapshot()
})
