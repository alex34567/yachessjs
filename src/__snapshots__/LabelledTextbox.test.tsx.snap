// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox with class 1`] = `
<div>
  <div
    class="LabelledTextbox"
  >
    <label>
      Hi
    </label>
    <input
      class="Test LabelledTextboxTextbox"
      type="text"
    />
  </div>
</div>
`;

exports[`Textbox without class 1`] = `
<div>
  <div
    class="LabelledTextbox"
  >
    <label>
      Hi
    </label>
    <input
      class="LabelledTextboxTextbox"
      type="text"
    />
  </div>
</div>
`;
